app/packages/addons-service/provider.ts

```diff
@@ -119,6 +120,14 @@ export default class Provider {
         this.getProviderConfiguredAddons();
     }
 
+    @task
+    @waitFor
+    async removeConfiguredAddon(selectedConfiguration: AllConfiguredAddonTypes) {
+        this.configuredAddons?.removeObject(selectedConfiguration);
+        notifyPropertyChange(this, 'configuredAddons');
+        await selectedConfiguration?.destroyRecord();
```

Comment:
Minor nit, but we may want to `destroyRecord` first and then remove it from the array after that has succeeded. May also be nice to add a try-catch in here and toast on success and errors

Reply:
Done.